jetbrains_plugin/Architecture.md

```
@@ -0,0 +1,871 @@
+1. **扩展提供者注册机制**：
+   
+   ```kotlin
+   private fun registerExtensionProviders() {
+       // 创建 Roo Code 扩展提供者实例
+       val rooProvider = com.sina.weibo.agent.extensions.roo.RooExtensionProvider()
+       registerExtensionProvider(rooProvider)
+   
+       // 创建 Cline AI 扩展提供者实例  
+       val clineProvider = com.sina.weibo.agent.extensions.cline.ClineExtensionProvider()
+       registerExtensionProvider(clineProvider)
+   }
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)